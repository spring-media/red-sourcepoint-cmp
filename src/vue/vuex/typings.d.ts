import { CustomPurpose, CustomVendor } from '../../sourcepoint/typings';

export type SourcepointModuleState = {
  consentedCustomVendors: CustomVendor[];
  consentedCustomPurposes: CustomPurpose[];
  grantedVendors: string[];
  consentReady: boolean;
};
