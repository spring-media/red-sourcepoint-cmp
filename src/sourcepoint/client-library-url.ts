export const getClientLibraryUrl = (): string => `https://cdn.privacy-mgmt.com/wrapperMessagingWithoutDetection.js`;
