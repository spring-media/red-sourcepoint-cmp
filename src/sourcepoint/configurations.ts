import { ConfigurationParameters, SourcepointConfigurationObject } from './typings';

export const createConfiguration = (parameters: ConfigurationParameters): Partial<SourcepointConfigurationObject> => ({
  config: {
    accountId: parameters.accountId,
    wrapperAPIOrigin: 'https://wrapper-api.sp-prod.net/tcfv2',
    propertyId: parameters.propertyId,
    mmsDomain: 'https://message.sp-prod.net',
    events: parameters.events,
  },
});
