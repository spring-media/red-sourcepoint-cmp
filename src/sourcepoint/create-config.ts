import { Config } from './typings';

export const createConfig = (parameters: Config): Config => ({
  accountId: 0,
  wrapperAPIOrigin: 'https://wrapper-api.sp-prod.net/tcfv2',
  propertyId: 0,
  mmsDomain: 'https://message.sp-prod.net',
  ...parameters,
});
