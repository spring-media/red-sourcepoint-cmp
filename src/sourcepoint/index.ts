export * from './iab-stub-script';
export * from './client-library-url';
export * from './has-custom-consent';
export * from './privacy-manager';
export * from './create-config';
export * from './get-custom-vendor-consents';
