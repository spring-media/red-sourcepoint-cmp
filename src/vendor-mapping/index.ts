export * from './vendors';
export * from './purposes';
export * from './api';
