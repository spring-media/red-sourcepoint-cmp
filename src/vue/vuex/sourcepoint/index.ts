import { Commit, Dispatch, Module } from 'vuex';
import { getCustomVendorConsents } from '../../../sourcepoint';
import { SourcepointModuleState } from '../typings';
import { CustomPurpose, CustomVendor } from '../../../sourcepoint/typings';
import {
  configureGrants,
  configureVendorPurposeMapping,
  getGrantedVendors,
  loadVendorPurposeMapping,
} from '../../../vendor-mapping';

const moduleState: SourcepointModuleState = {
  consentedCustomVendors: [],
  consentedCustomPurposes: [],
  grantedVendors: [],
};

export const mutations = {
  setCustomVendorConsents(state: SourcepointModuleState, payload: CustomVendor[]): void {
    state.consentedCustomVendors = payload;
  },
  setCustomPurposeConsents(state: SourcepointModuleState, payload: CustomPurpose[]): void {
    state.consentedCustomPurposes = payload;
  },
  setGrantedVendors(state: SourcepointModuleState, payload: string[]): void {
    state.grantedVendors = payload;
  },
};

export const getters = {
  vendorHasConsent: (state: SourcepointModuleState) => (vendorId: string): boolean =>
    state.grantedVendors.includes(vendorId),
};

export const actions = {
  async refreshCustomVendorConsents({ commit }: { commit: Commit }): Promise<void> {
    const { consentedPurposes = [], consentedVendors = [], grants = {} } = await getCustomVendorConsents();

    configureGrants(grants);

    commit('setCustomVendorConsents', consentedVendors);
    commit('setCustomPurposeConsents', consentedPurposes);
    commit('setGrantedVendors', getGrantedVendors());
  },
  async bootstrapConsents({ dispatch }: { dispatch: Dispatch }, payload: { propertyId: number }): Promise<void> {
    await dispatch('refreshCustomVendorConsents');

    const mappings = await loadVendorPurposeMapping(payload.propertyId);

    configureVendorPurposeMapping(mappings);
  },
};

export const sourcepoint: Module<SourcepointModuleState, Record<string, unknown>> = {
  namespaced: true,
  state: moduleState,
  mutations,
  actions,
  getters,
};
