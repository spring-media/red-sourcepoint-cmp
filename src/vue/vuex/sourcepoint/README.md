# sourcepoint Vuex Module

## Effects

### reloadPageOnReject

This effect reloads the page after a vendor or purpose has been rejected.

## Example

```javascript
import Vue from 'vue';
import Vuex from 'vuex';
import { sourcepoint } from '@spring-media/red-sourcepoint-cmp/dist/esm/vue/vuex/sourcepoint';
import { reloadPageOnReject } from '@spring-media/red-sourcepoint-cmp/dist/esm/vue/vuex/sourcepoint/effects';

Vue.use(Vuex);

const store = new Vuex.Store({
  strict: true,
  state: {},
  modules: {
    sourcepoint,
  },
});

reloadPageOnReject(store);
``` 
