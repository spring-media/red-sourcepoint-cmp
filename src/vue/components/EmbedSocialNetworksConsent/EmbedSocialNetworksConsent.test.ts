import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import { EmbedSocialNetworksConsent } from './';
import { sourcepoint } from '../../vuex/sourcepoint';
import { PURPOSE_ID_SOCIAL, VENDOR_ID_FACEBOOK } from '../../../vendor-mapping';

const localVue = createLocalVue();

localVue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    sourcepoint,
  },
});

describe('EmbedSocialNetworksConsent component', () => {
  afterEach(() => {
    store.commit('sourcepoint/setCustomVendorConsents', []);
    store.commit('sourcepoint/setCustomPurposeConsents', []);
  });

  it('should render the placeholder by default', () => {
    const wrapper = mount(EmbedSocialNetworksConsent, {
      propsData: {
        vendorId: 'abc',
        privacyManagerId: 12345,
      },
      store,
      localVue,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render the embed in case of a consented vendor and his purpose(s)', async () => {
    store.commit('sourcepoint/setCustomVendorConsents', [{ _id: VENDOR_ID_FACEBOOK }]);
    store.commit('sourcepoint/setCustomPurposeConsents', [{ _id: PURPOSE_ID_SOCIAL }]);

    const wrapper = mount(EmbedSocialNetworksConsent, {
      propsData: {
        vendorId: VENDOR_ID_FACEBOOK,
        privacyManagerId: 12345,
        content: '<div>Social Network Embed Content</div>',
      },
      store,
      localVue,
    });

    await wrapper.vm.$nextTick();

    expect(wrapper.element).toMatchInlineSnapshot(`
      <div>
        <div>
          Social Network Embed Content
        </div>
      </div>
    `);
  });
});
