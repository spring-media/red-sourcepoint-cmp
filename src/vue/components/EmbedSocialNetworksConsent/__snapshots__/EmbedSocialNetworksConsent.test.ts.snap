// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedSocialNetworksConsent component should render the embed in case of a consented vendor 1`] = `
<div>
  <div>
    <div>
      Social Network Embed Content
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`EmbedSocialNetworksConsent component should render the placeholder by default 1`] = `
<div
  class="embed-placeholder__container"
  data-vendor-id="abc"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="80"
      viewBox="0 0 102 80"
      width="102"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        opacity="0.3"
      >
        <path
          d="M65.9658 42.7696C68.8825 39.1697 73.3392 36.8686 78.3333 36.8686C83.3274 36.8686 87.7841 39.1697 90.7008 42.7696C89.9041 43.8122 88.9786 44.751 87.948 45.5624C85.7549 42.6446 82.2645 40.7575 78.3333 40.7575C74.4021 40.7575 70.9117 42.6446 68.7186 45.5624C67.688 44.751 66.7625 43.8122 65.9658 42.7696Z"
          fill="#CED4DA"
        />
         
        <path
          clip-rule="evenodd"
          d="M78.3333 35.1009C82.2384 35.1009 85.404 31.9352 85.404 28.0302C85.404 24.1251 82.2384 20.9595 78.3333 20.9595C74.4283 20.9595 71.2626 24.1251 71.2626 28.0302C71.2626 31.9352 74.4283 35.1009 78.3333 35.1009ZM78.3333 31.212C80.0906 31.212 81.5151 29.7875 81.5151 28.0302C81.5151 26.2729 80.0906 24.8484 78.3333 24.8484C76.576 24.8484 75.1515 26.2729 75.1515 28.0302C75.1515 29.7875 76.576 31.212 78.3333 31.212Z"
          fill="#CED4DA"
          fill-rule="evenodd"
        />
      </g>
       
      <g
        opacity="0.5"
      >
        <path
          d="M11.2991 42.7696C14.2158 39.1697 18.6724 36.8686 23.6666 36.8686C28.6607 36.8686 33.1174 39.1697 36.0341 42.7696C35.2374 43.8122 34.3119 44.751 33.2813 45.5624C31.0881 42.6446 27.5977 40.7575 23.6666 40.7575C19.7354 40.7575 16.245 42.6446 14.0519 45.5624C13.0213 44.751 12.0958 43.8122 11.2991 42.7696Z"
          fill="#CED4DA"
        />
         
        <path
          clip-rule="evenodd"
          d="M23.6666 35.1009C27.5716 35.1009 30.7373 31.9352 30.7373 28.0302C30.7373 24.1251 27.5716 20.9595 23.6666 20.9595C19.7615 20.9595 16.5959 24.1251 16.5959 28.0302C16.5959 31.9352 19.7615 35.1009 23.6666 35.1009ZM23.6666 31.212C25.4238 31.212 26.8484 29.7875 26.8484 28.0302C26.8484 26.2729 25.4238 24.8484 23.6666 24.8484C21.9093 24.8484 20.4848 26.2729 20.4848 28.0302C20.4848 29.7875 21.9093 31.212 23.6666 31.212Z"
          fill="#CED4DA"
          fill-rule="evenodd"
        />
      </g>
       
      <path
        clip-rule="evenodd"
        d="M62.9892 30.7877C62.9892 37.4821 57.5624 42.9089 50.868 42.9089C44.1737 42.9089 38.7468 37.4821 38.7468 30.7877C38.7468 24.0934 44.1737 18.6665 50.868 18.6665C57.5624 18.6665 62.9892 24.0934 62.9892 30.7877ZM50.868 45.9392C42.3066 45.9392 35.3333 49.8841 30.3333 56.0553L34.3857 58.1763C38.1453 53.1744 44.1289 49.9392 50.868 49.9392C57.6072 49.9392 63.5736 53.1744 67.3333 58.1763L71.3333 56.0553C66.3332 49.8841 59.4294 45.9392 50.868 45.9392ZM58.8888 30.6665C58.8888 35.0848 55.3071 38.6665 50.8888 38.6665C46.4705 38.6665 42.8888 35.0848 42.8888 30.6665C42.8888 26.2482 46.4705 22.6665 50.8888 22.6665C55.3071 22.6665 58.8888 26.2482 58.8888 30.6665Z"
        fill="#CED4DA"
        fill-rule="evenodd"
      />
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
     An dieser Stelle findest du Inhalte aus sozialen Netzwerken 
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
      Um mit Inhalten aus sozialen Netzwerken zu interagieren oder diese darzustellen, brauchen wir deine Zustimmung.
    
  </div>
   
  <button
    class="embed-placeholder__button"
  >
     soziale Netzwerke aktivieren 
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
        Ich bin damit einverstanden, dass mir Inhalte von Drittanbietern angezeigt werden. Damit können
        personenbezogene Daten an Drittanbieter übermittelt werden. Dazu ist ggf. die Speicherung von Cookies auf
        deinem Gerät notwendig. Mehr Informationen dazu findest du
        
    <a
      class="embed-placeholder__text-link"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      hier
    </a>
    .
      
  </div>
</div>
`;
